connector/connect/common/src/main/scala/org/apache/spark/sql/connect/common/DataTypeProtoConverter.scala

```diff
@@ -80,7 +81,7 @@ object DataTypeProtoConverter {
   }
 
   private def toCatalystStringType(t: proto.DataType.String): StringType =
-    StringType(t.getCollationId)
+    StringType(if (t.getCollation.nonEmpty) t.getCollation else "UTF8_BINARY")
```

Comment:
Why would the collation ever be empty?

Reply:
Please check protobuf [comment](https://github.com/apache/spark/pull/46180#discussion_r1601306785). The goal here is to support backwards compatibility in case of old PySpark client version for example. From protobuf [docs](https://protobuf.dev/programming-guides/proto3/#default), the default value for string is empty string so if client fails to set this field (in case of old client), we interpret collation as `UTF8_BINARY`.